src/zenml/new/steps/log_artifact_metadata.py

```diff
@@ -0,0 +1,120 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Utility functions for logging artifact metadata."""
+
+from typing import Dict, Optional
+
+from zenml.exceptions import StepContextError
+from zenml.metadata.metadata_types import MetadataType
+from zenml.new.steps.step_context import get_step_context
+
+
+def log_artifact_metadata(
+    output_name: Optional[str] = None,
+    description: Optional[str] = None,  # TODO: Do we really need this?
+    **kwargs: MetadataType,
+) -> None:
+    """Log artifact metadata.
+
+    Args:
+        output_name: The output name of the artifact to log metadata for. Can
+            be omitted if there is only one output artifact.
+        description: A description of the artifact.
+        **kwargs: Other metadata to log.
+
+    Raises:
+        RuntimeError: If the function is called outside of a step.
+        ValueError: If the function is called outside of a step.
+    """
+    kwargs = kwargs or {}
+
+    if description:
+        kwargs["description"] = description
+
+    try:
+        step_context = get_step_context()
+    except StepContextError:
+        raise RuntimeError("Cannot log artifact metadata outside of a step.")
+
+    try:
+        step_context.add_output_metadata(output_name=output_name, **kwargs)
+    except StepContextError as e:
+        raise ValueError(e)
+
+
+# TODO: Do we really need this?
+def log_model_object_metadata(
```

Comment:
Yes, @stefannica, you are right. These 2 are actually for consistency on the UI side. Imagine that users log `performance_metrics` instead of `metrics` or `predict_link` instead of `predict_url` - it would be tricky to handle on the UI (impossible). But with these wrappers, we give guidance/hints about what we expect to see in this kind of logging. WDYT?

Reply:
Yeah I think that makes sense, let's keep them. We're merging into the watch tower PR anyways, so we can still adjust those functions in the end in case the "special" metadata fields change.

Reply:
Fine by me